core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -456,13 +456,22 @@ public DefaultFileSystemMaster(BlockMaster blockMaster, CoreMasterContext master
 
   private static MountInfo getRootMountInfo(MasterUfsManager ufsManager) {
     try (CloseableResource<UnderFileSystem> resource = ufsManager.getRoot().acquireUfsResource()) {
+      boolean shared;
+      if (resource.get().isObjectStorage()) {
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.UNDERFS_OBJECT_STORE_MOUNT_SHARED_PUBLICLY);
+      } else {
+        shared = ServerConfiguration.getBoolean(
+                PropertyKey.MASTER_MOUNT_TABLE_ROOT_SHARED);
+      }
```

Comment:
I tried to update `shared` to this logic which seems more correct. And subsequently found issue https://github.com/Alluxio/alluxio/issues/10896
The issue related to `shared` seems deeper than I thought. I will revert the change regarding `shared` field and focus on `readonly` in this PR. I will attempt on the `shared` field in a separate PR later.